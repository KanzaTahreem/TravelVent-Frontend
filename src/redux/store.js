import { configureStore } from '@reduxjs/toolkit';
import cartItems from './cartSlice';
import registrationSlice from './reducers/registrationSlice';

const store = configureStore({
  reducer: {
    cartItems,
    registration: registrationSlice,
  },
});

export default store;
