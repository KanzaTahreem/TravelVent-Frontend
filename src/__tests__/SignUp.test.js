import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { screen, waitFor } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store';
import SignUp from '../pages/Signup';

describe('Testing SignUp', () => {
  let signup;
  beforeAll(async () => {
    signup = renderer
      .create(
        <Provider store={store}>
          <Router>
            <SignUp />
          </Router>
        </Provider>,
      )
      .toJSON();
  });

  it('should renders correctly', () => {
    expect(signup).toBeTruthy();
  });

  it('should match the snapshot', () => {
    expect(signup).toMatchSnapshot();
  });

  it('Should have a button with sign up text', () => {
    const btn = screen.findByTestId('submit');
    waitFor(() => expect(btn).toHaveTextContent('Sign up'));
  });

  it('renders "Sign In" link', () => {
    const link = screen.findByTestId('Sign In');
    waitFor(() => expect(link).toBeInTheDocument());
    waitFor(() => expect(link).toHaveAttribute('href', '/sign-in'));
  });

  it('renders all form fields', () => {
    waitFor(() => expect(screen.getByLabelText('Full Name')).toBeInTheDocument());
    waitFor(() => expect(screen.getByLabelText('Email')).toBeInTheDocument());
    waitFor(() => expect(screen.getByLabelText('Password')).toBeInTheDocument());
    waitFor(() => expect(screen.getByLabelText('Confirm Password')).toBeInTheDocument());
  });
});
