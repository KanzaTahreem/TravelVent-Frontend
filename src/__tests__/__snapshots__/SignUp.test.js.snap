// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing SignUp should match the snapshot 1`] = `
<section
  className="session-page signup"
>
  <div
    className="over-lay"
  >
    <div
      className="form-container"
    >
      <div
        className="text-column"
      >
        <h1>
          Create an account now!
        </h1>
        <h2>
          Your next adventure starts here!
        </h2>
        <div
          className="error-handler"
        >
          <div>
            <span
              className="error-msg"
            />
             
            <span
              className="error-msg"
            />
            <p
              className="error-msg"
            />
          </div>
        </div>
        <form>
          <div
            className="form-group"
          >
            <label
              htmlFor="name"
            >
              <input
                id="name"
                name="name"
                onChange={[Function]}
                required="required"
                type="text"
                value=""
              />
              <span>
                Full Name
              </span>
            </label>
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="email"
            >
              <input
                id="email"
                name="email"
                onChange={[Function]}
                required="required"
                type="email"
                value=""
              />
              <span>
                Email
              </span>
            </label>
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="password"
            >
              <input
                name="password"
                onChange={[Function]}
                required="required"
                type="password"
                value=""
              />
              <span>
                Password
              </span>
              <button
                className="eye-btn"
                onClick={[Function]}
                type="button"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 1024 1024"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <defs>
                    <clipPath>
                      <path
                        clipRule="evenodd"
                        d="M124-288l388-672 388 672H124z"
                        fill="none"
                      />
                    </clipPath>
                  </defs>
                  <path
                    d="M508 624a112 112 0 0 0 112-112c0-3.28-.15-6.53-.43-9.74L498.26 623.57c3.21.28 6.45.43 9.74.43zm370.72-458.44L836 122.88a8 8 0 0 0-11.31 0L715.37 232.23Q624.91 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 0 0 0 51.5q56.7 119.43 136.55 191.45L112.56 835a8 8 0 0 0 0 11.31L155.25 889a8 8 0 0 0 11.31 0l712.16-712.12a8 8 0 0 0 0-11.32zM332 512a176 176 0 0 1 258.88-155.28l-48.62 48.62a112.08 112.08 0 0 0-140.92 140.92l-48.62 48.62A175.09 175.09 0 0 1 332 512z"
                  />
                  <path
                    d="M942.2 486.2Q889.4 375 816.51 304.85L672.37 449A176.08 176.08 0 0 1 445 676.37L322.74 798.63Q407.82 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 0 0 0-51.5z"
                  />
                </svg>
              </button>
            </label>
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="password"
            >
              <input
                name="password"
                onChange={[Function]}
                required="required"
                type="password"
                value=""
              />
              <span>
                Confirm Password
              </span>
              <button
                className="eye-btn"
                onClick={[Function]}
                type="button"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 1024 1024"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <defs>
                    <clipPath>
                      <path
                        clipRule="evenodd"
                        d="M124-288l388-672 388 672H124z"
                        fill="none"
                      />
                    </clipPath>
                  </defs>
                  <path
                    d="M508 624a112 112 0 0 0 112-112c0-3.28-.15-6.53-.43-9.74L498.26 623.57c3.21.28 6.45.43 9.74.43zm370.72-458.44L836 122.88a8 8 0 0 0-11.31 0L715.37 232.23Q624.91 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 0 0 0 51.5q56.7 119.43 136.55 191.45L112.56 835a8 8 0 0 0 0 11.31L155.25 889a8 8 0 0 0 11.31 0l712.16-712.12a8 8 0 0 0 0-11.32zM332 512a176 176 0 0 1 258.88-155.28l-48.62 48.62a112.08 112.08 0 0 0-140.92 140.92l-48.62 48.62A175.09 175.09 0 0 1 332 512z"
                  />
                  <path
                    d="M942.2 486.2Q889.4 375 816.51 304.85L672.37 449A176.08 176.08 0 0 1 445 676.37L322.74 798.63Q407.82 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 0 0 0-51.5z"
                  />
                </svg>
              </button>
            </label>
          </div>
          <button
            className="submit-btn"
            onClick={[Function]}
            type="submit"
          >
            <span
              data-testid="submit"
            >
              Sign up
            </span>
          </button>
        </form>
        <p>
          Already have an account?
          <a
            className="link"
            data-testid="link"
            href="/sign-in"
            onClick={[Function]}
          >
             Sign In
          </a>
        </p>
      </div>
      <div
        className="image"
      />
    </div>
  </div>
</section>
`;
